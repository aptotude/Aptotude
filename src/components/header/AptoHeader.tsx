import React from 'react';
import classNames from 'classnames';
import './aptoHeader.scss';
import { StandardTypes } from '../../utils/standardTypes';

interface Props extends StandardTypes {
  type?: '1' | '2' | '3' | '4' | '5' | '6';
}

const COMPONENT_PREFIX = 'AptoHeader';

class AptoHeader extends React.Component<Props> {
  public static defaultProps = {
    type: '1'
  };

  public render() {
    const {
      type,
      className,
      forwardRef,
      children,
      ...rest
    } = this.props;

    let Component: any = 'h1';

    switch (type) {
      case '2':
        Component = 'h2';
        break;
      case '3':
        Component = 'h3';
        break;
      case '4':
        Component = 'h4';
        break;
      case '5':
        Component = 'h5';
        break;
      case '6':
        Component = 'h6';
        break;
      case '1':
      default:
        Component = 'h1';
        break;
    }

    const classes = classNames(
      COMPONENT_PREFIX,
      type && `${COMPONENT_PREFIX}--h${type}`,
      className
    );

    return (
      <Component
        {...rest}
        ref={forwardRef}
        className={classes}>
          {children}
      </Component>
    );
  }
}

export default AptoHeader;