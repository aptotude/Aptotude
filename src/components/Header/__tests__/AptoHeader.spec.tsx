import React from 'react';
import renderer from 'react-test-renderer';
import AptoHeader from '../AptoHeader';

test('Header renders', () => {
  const component = renderer.create(
    <AptoHeader />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Header sizes render', () => {
  const component = renderer.create(
    <div>
        <AptoHeader />
        <AptoHeader type="1" />
        <AptoHeader type="2" />
        <AptoHeader type="3" />
        <AptoHeader type="4" />
        <AptoHeader type="5" />
        <AptoHeader type="6" />
    </div>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
