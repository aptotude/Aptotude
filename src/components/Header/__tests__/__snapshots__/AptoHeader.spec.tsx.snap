// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<h1
  className="AptoHeader AptoHeader--h1"
/>
`;

exports[`Header sizes render 1`] = `
<div>
  <h1
    className="AptoHeader AptoHeader--h1"
  />
  <h1
    className="AptoHeader AptoHeader--h1"
  />
  <h2
    className="AptoHeader AptoHeader--h2"
  />
  <h3
    className="AptoHeader AptoHeader--h3"
  />
  <h4
    className="AptoHeader AptoHeader--h4"
  />
  <h5
    className="AptoHeader AptoHeader--h5"
  />
  <h6
    className="AptoHeader AptoHeader--h6"
  />
</div>
`;
